homeassistant/helpers/condition.py

```diff
@@ -134,23 +141,20 @@ def if_and_condition(
 and_from_config = _threaded_factory(async_and_from_config)
 
 
-def async_or_from_config(
-    config: ConfigType, config_validation: bool = True
+async def async_or_from_config(
+    hass: HomeAssistant, config: ConfigType, config_validation: bool = True
 ) -> Callable[..., bool]:
     """Create multi condition matcher using 'OR'."""
     if config_validation:
         config = cv.OR_CONDITION_SCHEMA(config)
-    checks = None
+    checks = [
+        await async_from_config(hass, entry, False) for entry in config["conditions"]
```

Comment:
OK to do move this here?

Reply:
sure. If `async_from_config` has to become async, we have no choice. Makes me wonder if we should also make all conditions async ?

Reply:
OK, but maybe for another PR?